src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -44,7 +44,7 @@
    [:source       [:enum :aggregation :fields :breakout :native]]
    ;; a field clause that can be used to refer to this Field if this query is subsequently used as a source query.
    ;; Added by this middleware as one of the last steps.
-   [:field_ref {:optional true} mbql.s/FieldOrAggregationReference]])
+   [:field_ref {:optional true} mbql.s/Reference]])
```

Comment:
Slightly unrelated but I renamed this in the legacy MBQL schema because it was confusing